libs/libarchfpga/src/read_fpga_interchange_arch.cpp

```diff
@@ -163,6 +234,220 @@ struct ArchReader {
+                std::string mux_type_string = entry.first ? "mux_" : "passGate_";
+                name << mux_type_string;
+
+                R = get_corner_value(model.getOutputResistance(), "slow", "min");
+                name << "R" << std::scientific << R;
+
+                Cin = get_corner_value(model.getInputCapacitance(), "slow", "min");
+                name << "Cin" << std::scientific << Cin;
+
+                Cout = get_corner_value(model.getOutputCapacitance(), "slow", "min");
+                name << "Cout" << std::scientific << Cout;
+
+                if (entry.first) {
+                    Cint = get_corner_value(model.getInternalCapacitance(), "slow", "min");
+                    name << "Cinternal" << std::scientific << Cint;
+                }
+
+                Tdel = get_corner_value(model.getInternalDelay(), "slow", "min");
+                name << "Tdel" << std::scientific << Tdel;
+
+                switch_name = name.str();
+                type = entry.first ? SwitchType::MUX : SwitchType::PASS_GATE;
+            }
+
+            /* Should never happen */
+            if (switch_name == std::string(VPR_DELAYLESS_SWITCH_NAME)) {
+                archfpga_throw(arch_file_, __LINE__,
+                               "Switch name '%s' is a reserved name for VPR internal usage!", switch_name.c_str());
+            }
+
+            as.name = vtr::strdup(switch_name.c_str());
+            as.set_type(type);
+            as.mux_trans_size = as.type() == SwitchType::MUX ? 1 : 0;
+
+            as.R = R;
+            as.Cin = Cin;
+            as.Cout = Cout;
+            as.Cinternal = Cint;
+            as.set_Tdel(t_arch_switch_inf::UNDEFINED_FANIN, Tdel);
+
+            if (as.type() == SwitchType::SHORT || as.type() == SwitchType::PASS_GATE) {
+                as.buf_size_type = BufferSize::ABSOLUTE;
+                as.buf_size = 0;
+                as.power_buffer_type = POWER_BUFFER_TYPE_ABSOLUTE_SIZE;
+                as.power_buffer_size = 0.;
+            } else {
+                as.buf_size_type = BufferSize::AUTO;
+                as.buf_size = 0.;
+                as.power_buffer_type = POWER_BUFFER_TYPE_AUTO;
+            }
+        }
+    }
+
+    void process_segments() {
+        auto strList = ar_.getStrList();
+
+        // Segment names will be taken from wires connected to pips
+        // They are good representation for nodes
+        std::set<uint32_t> wire_names;
+        for (auto tile_type : ar_.getTileTypeList()) {
+            auto wires = tile_type.getWires();
+            for (auto pip : tile_type.getPips()) {
+                wire_names.insert(wires[pip.getWire0()]);
+                wire_names.insert(wires[pip.getWire1()]);
+            }
+        }
+        int num_seg = wire_names.size();
+        arch_->Segments.resize(num_seg);
+        uint32_t index = 0;
+        for (auto i : wire_names) {
+            // Use default values as we will populate rr_graph with correct values
```

Comment:
Does it mean that we will always require users to provide a RRGraph file so that we will run the VPR in a correct way?
If we deposit all the bi-directional segments, I am not sure how it will impact the type of RRGraph: 

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/f92cfebaff5235224fe2f970c694ebca6daf7e8c/vpr/src/route/rr_graph.h#L12-L18

If it will cause the graph type to be always BIDIR, will it cause any problem when we parse a UNIDIR routing resource graph from a file?

Just need a double check here and identify any expected errors.

Reply:
In general, the RR graph is provided already within the device database (which may as well be split in the future from the rest of the architecture data).

Rather than a complete VPR-like RR graph, the interchange rr graph is a collection of nodes and PIPs, and we are working towards an Interchange RR graph reader, to read in the information from the interchange device data and build the VPR RR graph.

The [interchange schema](https://github.com/chipsalliance/fpga-interchange-schema/blob/53e3feda4628e39e939ce8d2e4add72025809d42/interchange/DeviceResources.capnp#L285-L297) can specify the directionality of a PIP and therefore whether some nodes are `UNIDIR` or `BIDIR`.

For the time being, the implementation proposed is to handle a simpler case, which can surely be made more complex to handle different kinds of graphs later on.

Reply:
Make sense to me. I guess you have already tried some RRGraph which have both ``UNIDIR`` and ``BIDIR`` segments before. Are they always defined as ``GRAPH_BIDIR``? If so, I believe it is fine.

Reply:
Yeah, I need to double check, but for instance, the Series7 graph in SymbiFlow is `BIDIR`, but IIRC we treat all the segments as `BIDIR`, actually. We'll experiment on whether adding support also for `UNIDIR` segments in a graph defined as `BIDIR`